x-pack/plugin/relevance-search/src/test/java/org/elasticsearch/xpack/relevancesearch/query/RelevanceMatchQueryIntTests.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.relevancesearch.query;
+
+import org.elasticsearch.action.search.SearchRequestBuilder;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ESSingleNodeTestCase;
+import org.elasticsearch.xpack.relevancesearch.RelevanceSearchPlugin;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertFailures;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertSearchHits;
+import static org.hamcrest.Matchers.containsString;
+
+public class RelevanceMatchQueryIntTests extends ESSingleNodeTestCase {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> getPlugins() {
+        return List.of(RelevanceSearchPlugin.class);
+    }
+
+    public void testRelevanceMatchQuery() {
+        client().admin().indices().prepareCreate("index").get();
```

Comment:
TIL that Java tests are very verbose. :)

I wonder if we can move the common parts of preparing the data into `setUp` and `tearDown`, and only dynamically inject the map of fields we want to set up. Or perhaps it's too much work?

Reply:
Very. And I'd be interested to see how Elasticsearch team writes tests with "good practices".

Reply:
The DataStreamTestHelper contains a lot of good example of "best practices"

Reply:
Also, these tests are good candidates to be implemented using YAML REST tests/.

An example of such a test is here: https://github.com/elastic/elasticsearch/blob/d2f99f5baf0e3b0cc2c706c80a4f7bb69efe1649/modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/10_basic.yml